wip-trampoline.md

```diff
@@ -0,0 +1,780 @@
+
+Constrained nodes must store information about nodes that are outside of their network view to use as trampoline hops.
+The `node_update` gossip message contains the information needed to use nodes as trampoline hops.
+
+Trampoline nodes on the contrary should keep a full view of the network to be able to efficiently route payments.
+
+### Fees and CLTV estimation
+
+Trampoline nodes need to estimate a `cltv_expiry_delta` and `fee` that allows them to route to any other trampoline
+node while being competitive with other nodes. This is a great opportunity to incentivize nodes to open channels
+between each other to minimize the cost of trampoline hops. This is also a great opportunity for nodes to implement
+smart fee estimation algorithms as a competitive advantage.
+
+Nodes may be very conservative and advertise their worst case `fee` and `cltv_expiry_delta`, corresponding to the
+furthest node they can reach.
+
+On the contrary, nodes may apply statistical analysis of the network to find a lower `fee` and `cltv_expiry_delta`
+that would not allow them to reach all other trampoline nodes but would work for most cases. Such nodes may choose
+to route some payments at a loss to keep reliability high, attract more payments by building a good reliability
+reputation and benefit from an overall gain.
+
+Nodes may include some of their preferred neighbours in `node_update`, implying that they're able to route cheaply to
+these nodes. Payers may or may not use that information when building a trampoline route.
+
+Trampoline nodes may accept payments with a fee lower than what they advertised if they're still able to route the
+payment in an economically viable way (because they have a direct channel or a low-cost route to the next trampoline
+hop for example).
+
+That means that payers may choose to ignore advertised fees entirely if they think the fee/cltv they're using will
+still be able to route properly.
+
+See the [Fees and CLTV requirements](#fees-and-cltv-requirements) section for a detailed example of fees calculation.
+
+### The `node_update` Message
+
+The `node_update` message has the following structure:
+
+1. type: 267 (`node_update`)
+2. types:
+    * type: 2 (`node_id`)
+    * data:
+        * [`point`:`node_id`]
+    * type: 4 (`node_signature`)
+    * data:
+        * [`signature`:`node_signature`]
+    * type: 6 (`chain_hash`)
+    * data:
+        * [`chain_hash`:`chain_hash`]
+    * type: 8 (`update_data`)
+    * data:
+        * [`u32`:`timestamp`]
+        * [`u16`:`cltv_expiry_delta`]
+        * [`u32`:`fee_base_msat`]
+        * [`u32`:`fee_proportional_millionths`]
+
+It has a structure similar to the `channel_update` message but is channel agnostic.
+`Node_update`s should be relayed the same way `channel_update`s are relayed (staggered broadcast).
+
+Trampoline nodes should only accept and relay `node_update`s from nodes that have open channels with inbound and
```

Comment:
Why the mandatory both inbound and outbound ? Seems acceptable to only trampoline route upstream or downstream packets

Reply:
You're totally right, this sentence doesn't make any sense...a node cannot know another node's channels balance so it's impossible to figure out that a node doesn't have inbound or outbound capacity. Don't know why I wrote that :thinking: